### PR TITLE
only use major and minor version for `prep_env` spec

### LINKED ISSUE #2621: prepare_env fails to register a user-provided (existing) venv for use with *named_env*
@andre-merzky says that I should be able to use an existing Python virtual environment with the *named_env* feature of an RP Task as long as it is registered with `prepare_env`. When I try this, though, I get an error that looks like #2589.

I.e.
```
Pilot cannot start task:
Traceback (most recent call last):
  File "/home/rp/radical.pilot.sandbox/rp.session.8acf1360-da9b-11ec-801c-0242ac120004/pilot.0000/rp_install/lib/python3.9/site-packages/radical/pilot/agent/executing/popen.py", line 122, in work
    self._handle_task(task)
  File "/home/rp/radical.pilot.sandbox/rp.session.8acf1360-da9b-11ec-801c-0242ac120004/pilot.0000/rp_install/lib/python3.9/site-packages/radical/pilot/agent/executing/popen.py", line 327, in _handle_task
    tmp += self._get_task_env(task, launcher)
  File "/home/rp/radical.pilot.sandbox/rp.session.8acf1360-da9b-11ec-801c-0242ac120004/pilot.0000/rp_install/lib/python3.9/site-packages/radical/pilot/agent/executing/popen.py", line 680, in _get_task_env
    ret += '. %s\n' % launcher.get_task_named_env(td['named_env'])
  File "/home/rp/radical.pilot.sandbox/rp.session.8acf1360-da9b-11ec-801c-0242ac120004/pilot.0000/rp_install/lib/python3.9/site-packages/radical/pilot/agent/launch_method/base.py", line 201, in get_task_named_env
    ru.env_prep(environment=ru.env_read(src),
  File "/home/rp/radical.pilot.sandbox/rp.session.8acf1360-da9b-11ec-801c-0242ac120004/pilot.0000/rp_install/lib/python3.9/site-packages/radical/utils/env.py", line 52, in env_read
    with ru_open(fname, 'r') as fin:
  File "/home/rp/radical.pilot.sandbox/rp.session.8acf1360-da9b-11ec-801c-0242ac120004/pilot.0000/rp_install/lib/python3.9/site-packages/radical/utils/misc.py", line 946, in ru_open
    return open(*args, **kwargs)
FileNotFoundError: [Errno 2] No such file or directory: '/home/rp/radical.pilot.sandbox/rp.session.8acf1360-da9b-11ec-801c-0242ac120004/pilot.0000/env/rp_named_env.test-env.env'
```

A minimal reproduction is provided at https://github.com/eirrgang/scale-ms/commit/e9e99bd74baea6877e7aebedd9026b607b3fbad9

### PR DESCRIPTION
**Pull request by andre-merzky**
fix #2621

### CODE REVIEW

File: bin/radical-pilot-create-static-ve [hunk 1]
```diff
@@ -112,19 +112,22 @@ esac
...
+then
+    VERSION=$($PYTHON -V | cut -f 2 -d ' ' | cut -f 1,2 -d '.')
+fi
+if test -z "$VERSION"
+then
+    VERSION='3.8'
+fi
+
+# we only ever use major and minor version numbers, and will ignore subminor
+VERSION=$(echo $VERSION | cut -f 1,2 -d '.')
```
Comment (mtitov) [inline bin/radical-pilot-create-static-ve:125]: maybe move bottom line into if-statement?
```
if test -z "$VERSION"
then
    VERSION='3.8'
else
    # we only ever use major and minor version numbers, and will ignore subminor
    VERSION=$(echo $VERSION | cut -f 1,2 -d '.')
fi
```
but could be skipped

**Review by mtitov — APPROVED** [review]
have small comment/suggestion, but overall LGTM!

File: bin/radical-pilot-create-static-ve (hunk 1 above)
Comment (andre-merzky) [reply bin/radical-pilot-create-static-ve:125]: like to check if subminor is specified in the first place?  Counting characters is cumbersome in shell, but also, the `cut` command also works if version is empty, major only or major.minor, so this should be safe.

File: bin/radical-pilot-create-static-ve [hunk 2]
```diff
@@ -112,19 +112,22 @@ esac
 
 PYTHON=python3
 
+if test -z "$VERSION"
+then
+    VERSION=$($PYTHON -V | cut -f 2 -d ' ' | cut -f 1,2 -d '.')
+fi
+if test -z "$VERSION"
+then
+    VERSION='3.8'
```
[the remark this answers is not in the document]
Comment (andre-merzky) [reply bin/radical-pilot-create-static-ve:121]: Line 117 should usually work and we might not even need the additional test and default setting, so I don't think that is tripping the scalems test case (but don't know for sure unless we do some more logging for this)

[the remark this answers is not in the document]
Comment (eirrgang) [reply bin/radical-pilot-create-static-ve:121]: It seems like the default is likely to do more harm than good, at least for the `venv` path, unless a check is made to find out what the current interpreter version actually is.

[the remark this answers is not in the document]
Comment (andre-merzky) [reply bin/radical-pilot-create-static-ve:121]: The default was removed meanwhile

**Pull request merged**